ert_shared/ensemble_evaluator/entity/unix_step.py

```diff
@@ -70,27 +70,30 @@ def run_jobs(self, client: Client, run_path: Path):
                 ev_source=job.get_source(self._ee_id),
             )
 
+    async def _exec_transformation(self, input_, transmitter, runpath):
+        await input_.get_transformation().dump(
+            transmitter=transmitter,
+            mime=input_.get_mime(),
+            runpath=runpath,
+            location=input_.get_path(),
+        )
+
     def _load_and_dump_input(
         self,
         transmitters: Dict[int, RecordTransmitter],
         runpath: Path,
     ):
-        Path(runpath / _BIN_FOLDER).mkdir(parents=True)
-
         futures = []
+        path_base = runpath
         for input_ in self._step.get_inputs():
-            path_base = runpath / _BIN_FOLDER if input_.is_executable() else runpath
             futures.append(
-                transmitters[input_.get_name()].dump(
-                    path_base / input_.get_path(), input_.get_mime()
+                self._exec_transformation(
+                    input_,
+                    transmitters[input_.get_name()],
+                    path_base,
```

Comment:
Do you need the intermediary `_exec_transformation` here? Both `input_.get_transformation().dump()` and `_exec_transformation` should be coros.

Reply:
Right, I did that but reverted it back :thinking: